src/modules/roc_fec/writer.cpp

```diff
@@ -147,24 +148,28 @@ packet::PacketPtr Writer::make_repair_packet_(packet::seqnum_t n) {
 
     packet->set_data(data);
 
+    fill_packet_rtp_fields_(packet, cur_block_repair_sn_ + pack_n, (pack_n == 0));
+    fill_packet_fec_fields_(packet, (packet::seqnum_t)n_source_packets_ + pack_n);
+
+    return packet;
+}
+
+void Writer::fill_packet_rtp_fields_(const packet::PacketPtr& packet,
+                                     packet::seqnum_t sn, bool marker) {
     packet::RTP& rtp = *packet->rtp();
 
     rtp.source = source_;
-    rtp.seqnum = cur_block_repair_sn_ + n;
-    rtp.marker = (n == 0);
+    rtp.seqnum = sn;
+    rtp.marker = marker;
     rtp.payload_type = 123;
-
-    fill_packet_fec_id_(packet, (packet::seqnum_t)n_source_packets_ + n);
-
-    return packet;
 }
 
-void Writer::fill_packet_fec_id_(const packet::PacketPtr& packet,
-                                 packet::seqnum_t pack_n) {
+void Writer::fill_packet_fec_fields_(const packet::PacketPtr& packet,
+                                     packet::seqnum_t pack_n) {
     packet::FEC& fec = *packet->fec();
 
-    fec.blknum = cur_block_source_sn_;
-    fec.source_block_length = n_source_packets_;
+    fec.source_block_number = cur_sbn_;
+    fec.source_block_length = (packet::blknum_t)n_source_packets_;
```

Comment:
Hm, sorry, I misled you. I think the source_block_length should have size_t type, not blknum_t. It encodes the number of packets per block. This number does not wrap so it does not need a separate type to specify its size. The default type we use for various "number of objects" variables is size_t.

Reply:
Done